src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/MemoryExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Runtime.CompilerServices;
+
+#if !NET8_0_OR_GREATER
+using System.Runtime.InteropServices;
+#endif
+
+namespace System;
+
+internal static class MemoryExtensions
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static unsafe void Replace(this ReadOnlySpan<char> source, Span<char> destination, char oldValue, char newValue)
+    {
+#if NET8_0_OR_GREATER
+        source.Replace(destination, oldValue, newValue);
+#else
+        var length = source.Length;
+        if (length == 0)
+        {
+            return;
+        }
+
+        if (length > destination.Length)
+        {
+            throw new ArgumentException(SR.Destination_is_too_short, nameof(destination));
```

Comment:
Does this need to be localized if we don't expect users to hit this?

Reply:
A user could certainly see any exception that isn't caught.